ddtrace/debugging/_origin/span.py

```diff
@@ -223,12 +222,12 @@ def on_span_start(self, span: Span) -> None:
 
                 span.set_tag_str(f"_dd.code_origin.frames.{n}.file", filename)
                 span.set_tag_str(f"_dd.code_origin.frames.{n}.line", str(code.co_firstlineno))
-                try:
-                    (f,) = functions_for_code(code)
-                    span.set_tag_str(f"_dd.code_origin.frames.{n}.type", f.__module__)
-                    span.set_tag_str(f"_dd.code_origin.frames.{n}.method", f.__qualname__)
-                except ValueError:
-                    continue
+
+                # Get the module and function name from the frame and code object. In Python3.11+ qualname
+                # is available, otherwise we'll fallback to the unqualified name.
+                (mod, name) = frame.f_globals.get("__name__"), getattr(code, "co_qualname", code.co_name)
```

Comment:
Maybe worth splitting this up and doing a version check/try..catch? With this approach we always look up `code.co_name`, even when `co_qualname` is available